api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationSearchAdviserTest.java

```diff
@@ -0,0 +1,450 @@
+        assertNotNull(location1);
+        assertNotNull(location2);
+        assertNotNull(location3);
+        assertNotNull(location4);
+        assertNotNull(location5);
+    }
+
+    @Test
+    public void getDefaultLocation_adminShouldGetDefaultLocation() {
+        setDefaultLocationGP(location2);
+        authenticateAdmin();
+        assertAuthenticatedAdmin();
+
+        Location result = locationService.getDefaultLocation();
+        assertEquals(location2, result);
+    }
+
+    @Test
+    public void getDefaultLocation_nonAdminWithLocationPropertyShouldGetOnlyDefaultLocationIfItsAccessLocation() {
+        setDefaultLocationGP(location3);
+        authenticateUserWithLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        Location result = locationService.getDefaultLocation();
+        assertEquals(location3, result);
+    }
+
+    @Test
+    public void getDefaultLocation_nonAdminWithLocationPropertyShouldNotGetDefaultLocationIfItsNotAccessLocation() {
+        setDefaultLocationGP(location2);
+        authenticateUserWithLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        Location result = locationService.getDefaultLocation();
+        assertNull(result);
+    }
+
+    @Test
+    public void getDefaultLocation_nonAdminWithoutLocationPropertyShouldGetOnlyDefaultLocationIfItsSessionLocation() {
+        setDefaultLocationGP(location2);
+        authenticateUserWithoutLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        setSessionLocation(location2);
+
+        Location result = locationService.getDefaultLocation();
+        assertEquals(location2, result);
+
+    }
+
+    @Test
+    public void getDefaultLocation_nonAdminWithoutLocationPropertyShouldNotGetDefaultLocationIfItsNotSessionLocation() {
+        setDefaultLocationGP(location2);
+        authenticateUserWithoutLocationProperty();
+        assertAuthenticatedNormalUser();
+
+        setSessionLocation(location3);
+
+        Location result = locationService.getDefaultLocation();
+        assertNull(result);
+    }
+
+    @Test
+    public void getLocationByUuid_adminShouldGetAllLocations() {
+        authenticateAdmin();
+        assertAuthenticatedAdmin();
+
+        Location result1 = locationService.getLocationByUuid(location1.getUuid());
+        assertEquals(location1, result1);
+        Location result2 = locationService.getLocationByUuid(location2.getUuid());
+        assertEquals(location2, result2);
+        Location result3 = locationService.getLocationByUuid(location3.getUuid());
+        assertEquals(location3, result3);
+        Location result4 = locationService.getLocationByUuid(location4.getUuid());
+        assertEquals(location4, result4);
+        Location result5 = locationService.getLocationByUuid(location5.getUuid());
+        assertEquals(location5, result5);
+    }
+
+    @Test
+    public void getLocationByUuid_nonAdminWithLocationPropertyShouldGetOnlyAccessLocation() {
```

Comment:
How did you ensure that location1 is an accessible location for user?

Reply:
Make sure to answer this question for other use cases as well

Reply:
@suthagar23 are you sure that you're looking at the right test dataset? My PR is bundled with `LocationTestData.xml`, just for this test suite.

Reply:
Also added the assert for user location property.